### PR TITLE
Add security policy

### PR DESCRIPTION
**Pull request by Betree**
> Pay a reward, not a ransom.

---

Resolve https://github.com/opencollective/opencollective/issues/2375

This PR is a proposal for a security bounty policy, as discussed during our team retreat in Brussels.

It is inspired by other companies policies, such as the one from [YesWeHack](https://yeswehack.com/programs/yes-we-hack) (that we may want to use at some point if we want to actively look for hunters).

When merged we should also create the special `security@opencollective.com` email to handle reports (except if a better alternative is suggested).

I've made an (arbitrary) proposal for bounty amounts, please review them and suggest others if you think they're not right.

---

Also related: https://github.com/opencollective/opencollective-frontend/pull/2146 adds a standard `security.txt` on our website.

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+## Scope
+
+We **won't** accept reports made by testing on our production servers (https://opencollective.com).
+
+We provide staging servers with a very similar configuration on the following URLs:
+
+- API: https://api-staging.opencollective.com
+- Frontend: https://staging.opencollective.com
+- Invoices: https://invoices-staging.opencollective.com
+- Images: https://images.opencollective.com
```
Comment (znarf) [inline SECURITY.md]: Not the staging URL

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+- Any issues regarding single session features/management
+- RTLO and related issues
+
+## Process
+
+- Reporter submit a report
+- Open Collective Team...
+  - Confirm that the report has been received
+  - Tries to reproduce the issue
+  - Discuss results and possible impact to detemine the score (low/medium/high/critical) and bounty amount
```
Comment (znarf) [inline SECURITY.md]:
```suggestion
  - Discuss results and possible impact to determine the score (low/medium/high/critical) and bounty amount
```

File: SECURITY.md (hunk 1 above)
Comment (znarf) [inline SECURITY.md]:
```suggestion
- Images: https://images-staging.opencollective.com
```

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+- Reporter is rewarded with a bounty at this stage (if applicable)
+- Team work on a fix, verify it an pushes it
```
Comment (znarf) [inline SECURITY.md]:
```suggestion
- Team work on a fix, verify it and pushes it
```

File: SECURITY.md (hunk 1 above)
Comment (Betree) [reply SECURITY.md]: Good catch, thanks!

**Comment by Betree** [comment]
I made the following changes to the policy:

- Add instructions about how to go public
- Add exploitability requirement 

See related commits for detailed changes

**Comment by Betree** [comment]
Added PGP public key for secure contact

**Pull request merged**